### PR TITLE
Flexible temperature in chi0 applies.

### PR DESCRIPTION
**Pull request by mfherbst**

### CODE REVIEW

File: src/postprocess/chi0.jl [hunk 1]
```diff
@@ -134,6 +134,7 @@ function apply_χ0(ham, δV, ψ, occupation, εF, eigenvalues; droptol=0,
 
             # Sternheimer contributions. TODO add preconditioning here
             !(sternheimer_contribution) && continue
+            fnk = filled_occ * Smearing.occupation(model.smearing, (εmk-εF) / temperature)
```
Comment (antoine-levitt) [inline src/postprocess/chi0.jl:137]: Careful line below

Comment (antoine-levitt) [reply src/postprocess/chi0.jl:137]: Just remove occupation arg?

Comment (antoine-levitt) [reply src/postprocess/chi0.jl:137]: Here and in the other chi0 probably

Comment (mfherbst) [reply src/postprocess/chi0.jl:137]: Not sure what you mean by the "other" chi0.

Comment (antoine-levitt) [reply src/postprocess/chi0.jl:137]: I meant compute_chi0 but I see that's irrelevant (although we could have a custom temperature there as well?)

File: src/postprocess/chi0.jl [hunk 2]
```diff
@@ -153,9 +153,9 @@ function apply_χ0(ham, δV, ψ, occupation, εF, eigenvalues; droptol=0,
     end
 
     # Add variation wrt εF
-    if model.temperature > 0
-        ldos = LDOS(εF, basis, eigenvalues, ψ)
-        dos = DOS(εF, basis, eigenvalues)
+    if temperature > 0
+        ldos = LDOS(εF, basis, eigenvalues, ψ, T=temperature)
+        dos = DOS(εF, basis, eigenvalues, T=temperature)
```
Comment (antoine-levitt) [inline src/postprocess/chi0.jl:158]: homogenize the `T` argument as temperature over there as well?

Comment (mfherbst) [reply src/postprocess/chi0.jl:158]: good idea. That's been bothering me for quite some time.

**Pull request merged**